### PR TITLE
Yechezkel/may sql2

### PR DESCRIPTION
**Pull request by chezi-nikop**

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 1]
```diff
@@ -45,4 +45,25 @@ class OwnerDao(private val sql: DSLContext) {
...
+
+    /*fun getOwnerById(id: Long): OwnerData {
+        return sql.select(owner.name, owner.companyId, owner.employeeId)
+            .from(owner)
+            .where(owner.ownerId.equal(id))
+
+    }
+     */
+
+    fun getOwnerIdFromPetId(id: Long): Long {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:64]: what is it? :)

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 2]
```diff
@@ -45,4 +45,25 @@ class OwnerDao(private val sql: DSLContext) {
             .and(petsTable.companyId.eq(companyId))
             .fetchOne(ownerMapper)
     }
+
+    fun getOwnerById(id: Long): OwnerData? {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:49]: what about companyId?

File: src/main/kotlin/com/hibob/academy/resource/OwnerApi.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.hibob.academy.dao.OwnerDataInsert
+import com.hibob.academy.service.OwnerService
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+
+@Controller
+@Path("/api/owner")
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/OwnerApi.kt:11]: it's not related to our ex but overall looking good

File: src/main/kotlin/com/hibob/academy/resource/PetsApi.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+
+@Controller
+@Path("/api/pets")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+class PetsResource(private val petsService: PetsService) {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/PetsApi.kt:15]: missing our methods
1. Get pets by owner: Given an owner ID, return all the information of the pets this owner adopted.
2. Count pets by type: This API does not receive any information but return a map of the type to how many pets of that type we have in the DB

File: src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt [hunk 5]
```diff
@@ -22,6 +23,8 @@ class PetsDaoTest @Autowired constructor(private val sql: DSLContext)  {
     @AfterEach
     fun cleanup() {
         sql.deleteFrom(tablePets).where(tablePets.companyId.eq(companyId)).execute()
+        sql.deleteFrom(tablePets).where(tablePets.companyId.eq(otherCompanyId)).execute()
```
Comment (yaelblsch) [inline src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt:26]: you can change to a single delete where company id in list of your test companies

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 6]
```diff
@@ -45,4 +46,19 @@ class PetsDao(private val sql: DSLContext) {
...
             .execute()
+
+    fun getPetsByOwnerId(ownerId: Long, companyId: Long): List<PetData> =
+        sql.select(petTable.id, petTable.ownerId, petTable.name, petTable.type, petTable.companyId, petTable.dateOfArrival)
+            .from(petTable)
+            .where(petTable.companyId.eq(companyId))
+            .and(petTable.ownerId.eq(ownerId))
+            .fetch(petMapper)
+
+    fun countPetsByType(): Map<PetType, Int> {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:57]: what about companyId here?

File: src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt [hunk 7]
```diff
@@ -14,14 +14,16 @@ class PetsDaoTest @Autowired constructor(private val sql: DSLContext)  {
...
+    private val otherCompanyId = 2L
 
     val petTest1 = PetDataInsert(ownerId = 1L, name = "bi", type = PetType.DOG, companyId)
     val petTest2 = PetDataInsert(null, name = "bi", type = PetType.DOG, companyId)
 
     @BeforeEach
     @AfterEach
     fun cleanup() {
-        sql.deleteFrom(tablePets).where(tablePets.companyId.eq(companyId)).execute()
+        sql.deleteFrom(tablePets).execute()
```
Comment (mayhuberman) [inline src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt:25]: you should have a delete fun in the dao and call her from here

**Review by mayhuberman — APPROVED** [review]

**Pull request merged**